tests/foreman/ui/test_search.py

```diff
@@ -0,0 +1,175 @@
+"""Test classes for Search component
+
+:Requirement: Search
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: Search
+
+:Team: Endeavour
+
+:TestType: Functional
+
+:CaseImportance: Medium
+
+:Upstream: No
+"""
+
+from fauxfactory import gen_string
+import pytest
+
+
+def menu_search_should_find(search_func: callable, values: list[str]):
+    """Helper function that does a vertical menu search for each value in `values` list
+    and makes sure it finds a corresponding match.
+    """
+    for search_string in values:
+        results = search_func(search_string)
+        assert results, f'No result found for search string {search_string}!'
+        for result in results:
+            assert (
+                search_string.lower() in result.lower()
+            ), f'Search string {search_string} does not match result {result}!'
+
+
+def menu_search_should_not_find(search_func: callable, values: list[str]):
+    """Helper function that does a vertical menu search for each value in `values` list
+    and makes sure it does not find any match.
+    """
+    for search_string in values:
+        results = search_func(search_string)
+        assert not results, f'Search string {search_string} should not return any results!'
+
+
+@pytest.fixture(scope='module')
+def auditor_module_user(module_target_sat, module_org, module_location):
+    """User with Auditor role. This user has significantly limited menu access."""
+    auditor_role = module_target_sat.api.Role().search(query={'search': 'name="Auditor"'})[0]
+    password = gen_string('alphanumeric')
+    user = module_target_sat.api.User(
+        admin=False,
+        location=[module_location],
+        organization=[module_org],
+        role=[auditor_role],
+        password=password,
+    ).create()
+    user.password = password
+
+    yield user
+
+    user.delete()
+
+
+@pytest.mark.tier2
+def test_positive_vertical_navigation_search_end_to_end(
+    module_target_sat,
+    module_org,
+    module_location,
+    default_org,
+    default_location,
+    auditor_module_user,
+    test_name,
+):
+    """Test the search function of the vertical navigation menu.
+
+    :id: 87660a22-996b-11ee-b8a5-000c2989e153
+
+    :Setup:
+        Create a non-admin user with Auditor role.
+        Create a custom organization and location.
+
+    :Steps:
+        1. Perform a search in the vertical navigation menu with following inputs:
+            1a. valid characters
+            1b. invalid characters
+            1c. current user organization and location
+            1d. custom organization and location not assigned to the user
+        2. Perform the actions as:
```

Comment:
I think user should be parametrized. So you will get two tests collected, with their respective users and fields expected to be/not be found.

Reply:
Ok, I've refactored the test to be parametrized. Passing some user-specific data - mainly fixtures - was a bit head-scratcher, but in the end I hope it's still readable and not overly complex.